pilot/pkg/proxy/envoy/proxy_test.go

```diff
@@ -19,6 +19,8 @@ import (
 	"reflect"
 	"testing"
 
+	"go.uber.org/atomic"
```

Comment:
Please add a couple of unit tests for this. I know this section of the code is a PITA to test but we should be verifying that this solves the issue in unit tests not manually.